pageant/src/lib.rs

```diff
@@ -11,8 +11,14 @@
     clippy::panic
 )]
 
-#[cfg(windows)]
+#[cfg(all(windows, feature = "wmmessage", not(feature = "namedpipes")))]
```

Comment:
Thanks a lot!

The two implementations should definitely not be mutually exclusive. They could be both exported together, with neither being a "default", e.g.:

```rust
#[cfg(all(windows, feature = "wmmessage"))]
pub use pageant_impl::PageantStream as PageantWmMessageStream;

#[cfg(all(windows, feature = "namedpipes"))]
pub use pageant_impl_namedpipes::PageantStream as PageantNamedPipeStream;
```

I'd really like to see a common trait for both of them since it would be weird to have to choose a specific implementation at compile time unless you know exactly which PuTTY version to expect in advance

Reply:
There is a cost to maintaining both and why would one want to run an old pre-2020 Putty Pageant with potential security bugs in 2025? I think a compile time decision is good enough, and I would even evaluate to eventually remove the  WM_MESSAGE method.

Reply:
Because in reality, when you're writing an SSH client, you don't control what version of Putty the user might have installed.

Reply:
P.S. I don't mind doing it myself, it will just take a bit until I have time to work on it.